### DISCUSSION TITLE
BIRT 4.17  - How to control which points on a line graph will be shown

### DISCUSSION THREAD

**Discussion by eacostagicarotech** [eclipse-birt/birt#1917] [General]
In a line graph with many records or points to be displayed, the AXIS-X labels end up creating a constant black line that is difficult to read. I am trying to hide some AXIS-X labels while still maintaining the original curve that is shown in the graph.

For example, if I have 288 records I want to show the complete curve of those 288 records but in AXIS-X; i only show 12 labels

Do you have any idea how to perform this control, perhaps using the script module? Would an example of how to perform this control be possible?

I appreciate any help and guidance in advance.

I am attaching an example of how the line graph is displayed today.

![Captura de tela 2024-09-19 173835](https://github.com/user-attachments/assets/81c345c6-0e40-44b1-83a1-749ee3f7bfb4)

**Comment by speckyspooky** [comment]
I have no full overview of your target because you have so many data point at your chart.

One think could be to reduce your label at the X-axis.
The chart has a special version of own scripting methods.
Here are 2 links in hope you will find some hints for your plan:
- #1848 with details of the handling of chart-methods and the usage of global variables 
- [http://birtworld.blogspot.com/search/label/Charts](url)

With these hints I would see a way to handle your label topic.
Each label will be handled with there own kind of "printing method" and with y global variable you can develope a logic,
to print only each 10st label or a larger range.

**Comment by eacostagicarotech** [comment]
Good afternoon everyone, Thank you for the response and sorry for the delay in my response. I was out of the office. I will try to replicate what is proposed in each link and I will be updating here.
